operator-framework-core/src/main/java/io/javaoperatorsdk/operator/api/reconciler/dependent/managed/ManagedDependentResourceContext.java

```diff
@@ -70,43 +71,63 @@ public Optional put(Object key, Object value) {
+        .map(entry -> new DependentResourceInfo(entry.getKey(), entry.getValue()))
+        .collect(Collectors.toMap(DependentResourceInfo::name, Function.identity()));
+  }
+
+  private DependentResourceInfo getDependentResource(String name) {
+    var dependentResource = dependentResources.get(name);
+    if (dependentResource == null) {
+      throw new OperatorException("No dependent resource found with name: " + name);
+    }
+    return dependentResource;
   }
 
   /**
-   * Retrieve all the known {@link DependentResource} implementations
+   * Retrieve the {@link ReconcileResult}, if it exists, associated with the
+   * {@link DependentResource} associated with the specified name
    *
-   * @return a list of known {@link DependentResource} implementations
+   * @param name the name of the {@link DependentResource} for which we want to retrieve a
+   *        {@link ReconcileResult}
+   * @return an Optional containing the reconcile result or {@link Optional#empty()} if no such
+   *         result is available
    */
-  public List<DependentResource> getDependentResources() {
-    return dependentResources;
+  @SuppressWarnings("rawtypes")
+  public Optional<ReconcileResult> getReconcileResult(String name) {
+    return Optional.of(getDependentResource(name)).map(DependentResourceInfo::result);
   }
 
   /**
-   * Retrieve the dependent resource implementation associated with the specified resource type.
+   * Set the {@link ReconcileResult} for the specified {@link DependentResource} implementation.
    *
-   * @param resourceClass the dependent resource class for which we want to retrieve the associated
-   *        dependent resource implementation
-   * @param <T> the type of the resources for which we want to retrieve the associated dependent
-   *        resource implementation
-   * @return the associated dependent resource implementation if it exists or an exception if it
-   *         doesn't or several implementations are associated with the specified resource type
+   * @param name the name of the {@link DependentResource} for which we want to set the
+   *        {@link ReconcileResult}
+   * @param reconcileResult the reconcile result associated with the specified
+   *        {@link DependentResource}
    */
-  @SuppressWarnings("unchecked")
-  public <T extends DependentResource> T getDependentResource(Class<T> resourceClass) {
-    var resourceList =
-        dependentResources.stream()
-            .filter(dr -> dr.getClass().equals(resourceClass))
-            .collect(Collectors.toList());
-    if (resourceList.isEmpty()) {
-      throw new OperatorException(
-          "No dependent resource found for class: " + resourceClass.getName());
+  public void setReconcileResult(String name, ReconcileResult reconcileResult) {
+    getDependentResource(name).result = reconcileResult;
+  }
+
+  private static class DependentResourceInfo {
+    private final String name;
+    private final DependentResource dependent;
```

Comment:
not used. Is a public getter missing or should it be removed?
I would remove it because being able to retrieve DependentResource is not needed anymore thanks to named event source. This is fine to me as the user cannot do bad things (like calling reconcile) and we got a real managed mode.
If removed, then, I don't see why we should keep this wrapper around reconcile result. It will create one more instance per dependent/per reconcile to keep track of the reconcileResult while the hashmap could store it directly.

Reply:
Yes, I was wondering about that but I agree that we should remove it for now. It'll be easier to add it later than remove it once it's already out there…